src/redux/reducers/cartSlice.ts

```diff
@@ -0,0 +1,215 @@
+        },
+      );
+      return response.data;
+    } catch (err) {
+      const error = err as AxiosError;
+      return rejectWithValue(error.response?.data || error.message);
+    }
+  },
+);
+
+export const updateCarts = createAsyncThunk(
+  "update",
+  async (
+    { productId, quantity }: { productId: string; quantity: number },
+    { rejectWithValue },
+  ) => {
+    try {
+      const response = await axios.patch(
+        "/carts",
+        { productId, quantity },
+        {
+          headers: {
+            Authorization: `Bearer ${localStorage.getItem("accessToken")}`,
+          },
+        },
+      );
+      console.log(response);
+
+      return response.data;
+    } catch (err) {
+      const error = err as AxiosError;
+      return rejectWithValue(error.response?.data || error.message);
+    }
+  },
+);
+
+const cartManageSlice = createSlice({
+  name: "carts",
+  initialState: {
+    isLoading: false,
+    data: [] as UserCart[],
+    error: false,
+    delete: {
+      isLoading: false,
+      error: false,
+    },
+    add: {
+      isLoading: false,
+      data: [],
+      error: null,
+    },
+    remove: {
+      isLoading: false,
+      error: false,
+    },
+    update: {
+      isLoading: false,
+      data: [],
+      error: false,
+    },
+  },
+  reducers: {
+    increaseQuantity: (state, action: PayloadAction<string>) => {
+      const item = state.data.find((item) => item.id === action.payload);
+      if (item) {
+        item.quantity += 1;
+      }
+    },
+    decreaseQuantity: (state, action: PayloadAction<string>) => {
+      const item = state.data.find((item) => item.id === action.payload);
+      if (item && item.quantity > 1) {
+        item.quantity -= 1;
+      }
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(cartManage.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(cartManage.fulfilled, (state, action) => {
+        state.isLoading = false;
+        state.data = action.payload;
+      })
+      .addCase(cartManage.rejected, (state) => {
+        state.isLoading = false;
+        state.error = true;
+      })
+      .addCase(cartDelete.pending, (state) => {
+        state.delete.isLoading = true;
+      })
+      .addCase(cartDelete.fulfilled, (state) => {
+        state.delete.isLoading = false;
+      })
+      .addCase(cartDelete.rejected, (state) => {
+        state.delete.error = true;
+      })
+      .addCase(removeFromCart.pending, (state) => {
+        state.remove.isLoading = true;
+      })
+      .addCase(removeFromCart.fulfilled, (state) => {
+        state.remove.isLoading = false;
+      })
+      .addCase(removeFromCart.rejected, (state) => {
+        state.remove.error = true;
+      })
+      .addCase(addToCart.pending, (state) => {
+        state.add.isLoading = true;
+      })
+      .addCase(addToCart.fulfilled, (state, action) => {
+        state.add.isLoading = false;
+        state.data.push(action.payload);
```

Comment:
from here u don't have to push payload into cart bcz if product already exists in cart it will have not change mean u are falsely increasing cart while there is no new item added

Reply:
this is related to missing remove from cart button in productCard.tsx if that button is there this will work as expected but sinc eits missing it won't